howdoi/howdoi.py

```diff
@@ -432,6 +434,61 @@ def _get_cache_key(args):
     return str(args) + __version__
 
 
+def print_stash(stash_list = []):
+    if len(stash_list) == 0:
+        stash_list = ['\nSTASH LIST:']
+        commands = keep_utils.read_commands()
+
+        if commands is None or len(commands.items()) == 0:
+            print('No commands found in stash. Add a command with "howdoi -save <query>".')
+            return
+
+        for cmd, fields in commands.items():
+            # \033[4m\033[1m to bold and underline text.
+            stash_list.append(
+                '\033[4m\033[1m$ ' + fields['alias'] + '\033[0m\n\n' 
+                + fields['desc'] + '\n')
+    else:
+        stash_list = [(
+            '\033[4m\033[1m$ [' + str(i+1) + '] ' + x['fields']['alias'] + '\033[0m\n\n' + x['fields']['desc'] + '\n') 
+            for i, x in enumerate(stash_list)]
+    print(build_splitter('#').join(stash_list))
+
+
+def _get_stash_key(args):
+    stash_args = {}
+    ignore_keys = ['stash_save', 'stash_view', 'stash_remove', 'stash_empty', 'tags']
+    for key in args:
+        if not (key in ignore_keys):
+            stash_args[key] = args[key]
+    return str(stash_args)
+
+
+def remove_stash_command(cmd_key, title):
```

Comment:
perhaps you can call these functions `_stash_remove`, `_stash_save`, `_stash_empty`, `_stash_list` which will better map to the names we use externally (more readable). Also please use the leading `_` to show that these are internal functions. Also please group all the stash functions together.